File: Medium/167_Two_Sum_II_Input_Array_Is_Sorted.py
class Solution:
    def twoSum(self, a: List[int], t: int) -> List[int]:
        
        i, j = 0, len(a)-1
        while i < j:
            sum = a[i] + a[j]
            if sum == t:
                return [i+1, j+1]
            elif sum < t:
                i += 1
            else:
                j -= 1
        return [-1, -1]

File: Medium/167_Two_Sum_II_Input_Array_Is_Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& a, int t) {
        int i=0, j=a.size()-1;
        while(i < j)
        {
            int sum = a[i] + a[j];
            if(sum == t)
            {
                return {i+1, j+1};
            }else if(sum < t)
            {
                i++;
            }else {
                j--;
            }
        }
        return {-1, -1};
    }
};
